src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+        }
+
+        string NewLongValue()
+        {
+            var buf = new StringBuilder(2 * this.data.Length + 5);
+            int i = 0;
+            i = this.AppendHexDumpField(buf, i, MachineIdLen);
+            i = this.AppendHexDumpField(buf, i, ProcessIdLen);
+            i = this.AppendHexDumpField(buf, i, SequenceLen);
+            i = this.AppendHexDumpField(buf, i, TimestampLen);
+            i = this.AppendHexDumpField(buf, i, RandomLen);
+            Debug.Assert(i == this.data.Length);
+            return buf.ToString().Substring(0, buf.Length - 1);
+        }
+
+        int AppendHexDumpField(StringBuilder buf, int i, int length)
+        {
+            buf.Append(ByteBufferUtil.HexDump(this.data, i, length));
+            buf.Append('-');
+            i += length;
+            return i;
+        }
+
+        void Init()
+        {
+            int i = 0;
+            // machineId
+            Array.Copy(MachineId, 0, this.data, i, MachineIdLen);
+            i += MachineIdLen;
+
+            // processId
+            i = this.WriteInt(i, ProcessId);
+
+            // sequence
+            i = this.WriteInt(i, Interlocked.Increment(ref nextSequence));
+
+            // timestamp (kind of)
+            long ticks = Stopwatch.GetTimestamp();
+            long nanos = (ticks / Stopwatch.Frequency) * 1000000000;
+            long millis = (ticks / Stopwatch.Frequency) * 1000;
+            i = this.WriteLong(i, ByteBufferUtil.SwapLong(nanos) ^ millis);
+
+            // random
+            int random = ThreadLocalRandom.Value.Next();
+            this.hashCode = random;
+            i = this.WriteInt(i, random);
+
+            Debug.Assert(i == this.data.Length);
+        }
+
+        int WriteInt(int i, int value)
+        {
+            uint val = (uint)value;
+            this.data[i++] = (byte)(val >> 24);
+            this.data[i++] = (byte)(val >> 16);
+            this.data[i++] = (byte)(val >> 8);
+            this.data[i++] = (byte)value;
+            return i;
+        }
+
+        int WriteLong(int i, long value)
+        {
+            ulong val = (ulong)value;
+            this.data[i++] = (byte)(val >> 56);
+            this.data[i++] = (byte)(val >> 48);
+            this.data[i++] = (byte)(val >> 40);
+            this.data[i++] = (byte)(val >> 32);
+            this.data[i++] = (byte)(val >> 24);
+            this.data[i++] = (byte)(val >> 16);
+            this.data[i++] = (byte)(val >> 8);
+            this.data[i++] = (byte)value;
+            return i;
+        }
+
+        public override int GetHashCode()
+        {
+            return this.hashCode;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == this)
+            {
+                return true;
+            }
+
+            if (!(obj is DefaultChannelId))
+            {
+                return false;
+            }
+
+            return Array.Equals(this.data, ((DefaultChannelId)obj).data);
```

Comment:
this is Object.Equals in disguise -> no good for this. Let's use `StructuralComparisons.StructuralEqualityComparer.Equals(...)` for now. If we see it being a issue through profiling, we'll address that.